src/App.tsx

```diff
@@ -1,24 +1,193 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { deleteTodoOnServer, getTodos, updateTodoOnServer } from './api/todos';
+import { filterTodos } from './helpers/filterTodos';
 
-const USER_ID = 0;
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+
+import { Todo } from './types/Todo';
+
+import { FilterType } from './enums/FilterType';
+import { ErrorType } from './enums/ErrorType';
+import { OnChangeFunc } from './types/OnChangeFunc';
+import { AuthContext } from './contexts/AuthContext';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState(FilterType.All);
+  const [errorType, setErrorType] = useState(ErrorType.None);
+  const [isErrorShown, setIsErrorShown] = useState(false);
+  const [tempTodoTitle, setTempTodoTitle] = useState('');
+  const [isAllToggled, setIsAllToggled] = useState(false);
+  const [isClearCompleted, setIsClearCompleted] = useState(false);
+
+  const { id: userId = 0 } = useContext(AuthContext) || {};
+
+  useEffect(() => {
+    getTodos(userId)
+      .then((userTodos) => setTodos(userTodos))
+      .catch(() => {
+        setErrorType(ErrorType.Download);
+        setIsErrorShown(true);
+      });
+  }, []);
+
+  const activeTodosNum = useMemo(() => {
+    return todos.reduce((num, todo) => {
+      return todo.completed ? num : num + 1;
+    }, 0);
```

Comment:
it looks too complicated for this case , I would `todos.filter(({ completed }) => completed).length`